src/qonlinetranslator.cpp

```diff
@@ -269,6 +298,46 @@ bool QOnlineTranslator::isRunning() const
     return m_stateMachine->isRunning();
 }
 
+QString QOnlineTranslator::toJson() const
+{
+    QJsonObject translationOptions;
+    for (auto i = m_translationOptions.begin(); i != m_translationOptions.end(); ++i)
+    {
+        QJsonArray arr;
+        for (const auto &j : i.value())
+        {
+            arr.append(j.toJson());
+        }
+        translationOptions.insert(i.key(), arr);
+    }
+
+    QJsonObject examples;
+    for (auto i = m_examples.begin(); i != m_examples.end(); ++i)
+    {
+        QJsonArray arr;
+        for (const auto &j : i.value())
+        {
+            arr.append(j.toJson());
+        }
+        examples.insert(i.key(), arr);
+    }
+
+    QJsonObject object
+    {
+        {"source", m_source},
+        {"sourceTranslit", m_sourceTranslit},
+        {"sourceTranscription", m_sourceTranscription},
+        {"translation", m_translation},
+        {"translationTranslit", m_translationTranslit},
+        {"errorString", m_errorString},
+        {"translationOptions", translationOptions},
+        {"examples", examples}
```

Comment:
Since these values are not used further, it makes sense to move them here.
```suggestion
        {"translationOptions", qMove(translationOptions)},
        {"examples", qMove(examples)}
```